docs/rfcs/017-asset-modified-cleanup.md

```diff
@@ -0,0 +1,140 @@
+
+Finally, maxUnauth/Roles will need to be taken into account as storage in the bucket will have a prefix of `/open` or `/auth` based on this.
+
+#### Named query derivatives
+
+There are many different permutations of objects in named queries, so for now this is out of scope for cleanup.
+
+#### Update types
+
+There are a number of ways that an update to an asset can cause changes to stored derivatives that aren't tracked.  It can roughly be divided into 4 categories of change:
+
+- Delivery channel changed
+- Policy id changed
+- Roles changed
+- Policy data updated
+- Origin changes
+
+#### Delivery channels changed
+
+This becomes an issue when a delivery channel is removed from an asset, with the following implications:
+
+- iiif-img removed
+  - Stored iiif-img derivative needs to be removed
+  - `info.json` removed
+  - Asset can exist at both filename and `/original` path - though the `original` needs to be left if the file channel exists on the asset
+  - if `thumbs` not there, then `s.json` and thumbnails need removed as well 
+- Thumbs removed
+  - `info.json` removed
+  - If removing "thumbs" leaves "iiif-img":
+    - Do not touch `s.json`
+    - Do not touch `AssetApplicationMetadata`
+    - Leave system thumbs
+  - If removing "thumbs" doesn't leave "iiif-img":
+    - Delete `s.json`
+    - Delete `AssetApplicationMetadata`
+    - Delete all thumbs
+- iiif-av removed
+  - Timebased derivative removed
+  - Metadata removed
+- File removed
+  - The asset at origin should be removed if there's an asset on the `/original` path - should only be removed if `iiif-img` is not using it
+- None removed
+  - Nothing required
+
+#### Policy id changed
+
+This is that the delivery channel stays the same, but the id of the policy has changed.  It should be able to be detected by checking the delivery channel policy id in `before` against the current asset, with the following implications:
+
+- iiif-img changed
+  - `info.json` needs removed
+- Thumbs changed
+  - Thumbs need to be removed that are no longer required.
+  - `s.json` and asset application metadata should be updated - `s.json` should be updated by the reingest
+  - `info.json` removed
+- iiif-av changed
+  - Old transcode derivative removed if the file extension is no longer required
+- File changed
+  - The asset at origin should be removed if there's an asset on the `/original` path - should only be removed if `iiif-img` is not using it
+
+#### Roles changed
+
+This should only have an implication on the `info.json`, which would need to be removed
+
+#### Policy data updated
+
+The policy data being updated can be found from the date that the delivery channel policy was updated, after the `finished` date of the current asset itself. 
+
+- iiif-img changed
+  - `info.json` needs removed
+  - If it moves to a `use-original` policy, is there a need to remove the asset as well?
```

Comment:
Yes - if `use-original` then the generated JP2 derivative won't be used. Similarly if it was `use-original` and went to `default` we can remove the `/original` item (if not used by "file" channel)